packages/@aws-cdk/aws-vpclattice-alpha/lib/service.ts

```diff
@@ -0,0 +1,220 @@
+import { CfnService } from 'aws-cdk-lib/aws-vpclattice/lib/vpclattice.generated';
+import { Resource, IResource, ValidationError, ArnFormat, Stack } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { ICertificate } from 'aws-cdk-lib/aws-certificatemanager';
+import { PolicyDocument } from 'aws-cdk-lib/aws-iam';
+
+/**
+ * Authentication types for VPC Lattice services
+ */
+export enum ServiceAuthType {
+  /**
+   * No authentication
+   */
+  NONE = 'NONE',
+
+  /**
+   * AWS IAM authentication
+   */
+  AWS_IAM = 'AWS_IAM',
+}
+
+/**
+ * DNS entry information for a VPC Lattice service
+ */
+export interface DnsEntry {
+  /**
+   * The domain name of the DNS entry
+   *
+   * @default - No domain name
+   */
+  readonly domainName?: string;
+
+  /**
+   * The hosted zone ID of the DNS entry
+   *
+   * @default - No hosted zone ID
+   */
+  readonly hostedZoneId?: string;
```

Comment:
Not [IHostedZone](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_route53.IHostedZone.html)?

Should the whole of `DnsEntry` just be a hosted zone?